### ISSUE TITLE
Comparison and boolean are not associative, grouping is necessary

### ISSUE THREAD

**Issue by mMerlin** [jshint/jshint#2280]
jshint.com says:

```
2 Unnecessary grouping operator
```

But the results are not the same without forcing the or operation to be done first.

```javascript
/* jshint singleGroups: true */
console.log(1 < (undefined || 1));
// ==> false
console.log(1 < undefined || 1);
// ==> 1
```

**Comment by jugglinmike** [comment]
Thanks for the report! This bug is addressed in gh-2270

**Comment by jugglinmike** [comment]
gh-2270 has been merged and released in JSHint 2.7.0 . Please let us know if this is still a problem for you!

**Comment by mMerlin** [comment]
Correct results with my simple test case at least
